controllers/sr_controller/sr_controller.py

```diff
@@ -176,10 +179,28 @@ class SimpleTee:
     Forwards calls from its `write` and `flush` methods to each of the given targets.
     """
 
-    def __init__(self, *streams: IO[str]) -> None:
+    def __init__(self, *streams: IO[str], prefix: str = '') -> None:
         self.streams = streams
+        self._line_start = True
+        self.prefix = prefix
+
+    def _insert_prefix(self, data: str) -> str:
+        # Append our prefix just after all inner newlines. Don't append to a
+        # trailing newline as we don't know if the next line in the log will be
+        # from this zone.
+        final_newline = data.endswith('\n')
```

Comment:
Do we know how this performs on windows, given that generally uses `\r\n` for newlines

Reply:
I'm assuming you mean this question about the following line (where we do the `replace`) rather than this line? Obviously this line will be completely fine on Windows.
---
So Python uses universal newlines -- any time it reads in or wants to handle a newline you _should_ get `\n` rather than `\r\n`. Obviously this doesn't stop someone putting a literal `\r\n` in their code and printing that and in that case this will likely get weird, however that code is then not portable and would have weird output on Unix anyway.